### PR TITLE
AN-547/hls timed metadata docs

### PR DESCRIPTION
**Pull request by jashan1498**

### CODE REVIEW

File: docs/android/v2/advanced-features/hls-timed-metadata.mdx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+## Requirements
+
+- 100ms Android SDK version 2.5.6 or higher
+- Active HLS stream
+
+## Sending HLS Timed Metadata
+
+To add HLS timed metadata cue to the currently running HLS stream use `setHlsSessionMetadata` API like this:
+
+```kotlin
```
Comment (arjunattam) [inline docs/android/v2/advanced-features/hls-timed-metadata.mdx:17]: do we need java examples?

File: docs/android/v2/advanced-features/hls-timed-metadata.mdx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+val metadata = HMSHLSTimedMetadata(
+                        payload  = payloadText,
+                        duration = durationText
+                    )
+
+hmsSDK.setHlsSessionMetadata(arrayListOf(metaDataModel), object : HMSActionResultListener {
```
Comment (arjunattam) [inline docs/android/v2/advanced-features/hls-timed-metadata.mdx:23]: is this correct? seeing `SessionMetadata`

File: docs/android/v2/advanced-features/hls-timed-metadata.mdx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+---
+ title: HLS Timed Metadata
+ nav: 8.05
+---
+
+HLS Timed Metadata feature helps you synchronise certain events with the HLS stream. This can be useful for showing interactive quizzes / product overlays etc.
+
+## Requirements
```
Comment (arjunattam) [inline docs/android/v2/advanced-features/hls-timed-metadata.mdx:8]: is there any requirements for exoplayer version?

File: docs/android/v2/advanced-features/hls-timed-metadata.mdx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+})
+```
+
+## Receiving HLS Timed Metadata
+
+To receive metadata on the HLS player side use `HlsMetadataHandler` class from sample app.
+
+```kotlin
+
+playerEventsManager = PlayerEventsCollector(it)
```
Comment (arjunattam) [inline docs/android/v2/advanced-features/hls-timed-metadata.mdx:40]: where is `it` coming from?

File: docs/android/v2/advanced-features/hls-timed-metadata.mdx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+val hlsMetadataHandler = HlsMetadataHandler(exoPlayer = exoplayer,{ metaDataModel ->
+ // showing a view or hiding it can be handled here
+} ,context)
```
Comment (arjunattam) [inline docs/android/v2/advanced-features/hls-timed-metadata.mdx:43]: spacing in the snippet seems a bit odd?

File: docs/android/v2/advanced-features/hls-timed-metadata.mdx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+hlsMetadataHandler.start()
+
+```
+
+Please note that callback will be fired every second if a matching tag is found, the callback firing rate can be modified directly in `HlsMetadataHandler` based on preference.
```
Comment (arjunattam) [inline docs/android/v2/advanced-features/hls-timed-metadata.mdx:48]: can we document how that's to be modified?

File: docs/android/v2/advanced-features/hls-timed-metadata.mdx (hunk 3 above)
Comment (jashan1498) [reply docs/android/v2/advanced-features/hls-timed-metadata.mdx:8]: nope

File: docs/android/v2/advanced-features/hls-timed-metadata.mdx (hunk 6 above)
Comment (jashan1498) [reply docs/android/v2/advanced-features/hls-timed-metadata.mdx:48]: we have already added a toast example for that

File: docs/android/v2/advanced-features/hls-timed-metadata.mdx (hunk 5 above)
Comment (jashan1498) [reply docs/android/v2/advanced-features/hls-timed-metadata.mdx:43]: updated

File: docs/android/v2/advanced-features/hls-timed-metadata.mdx (hunk 4 above)
Comment (jashan1498) [reply docs/android/v2/advanced-features/hls-timed-metadata.mdx:40]: resolved

File: docs/android/v2/advanced-features/hls-timed-metadata.mdx (hunk 2 above)
Comment (jashan1498) [reply docs/android/v2/advanced-features/hls-timed-metadata.mdx:23]: `SessionMetadata` where did you see this ?

File: docs/android/v2/advanced-features/hls-timed-metadata.mdx (hunk 6 above)
Comment (saikatmitra91) [reply docs/android/v2/advanced-features/hls-timed-metadata.mdx:75]: I think the ask is to document how part of 

> the callback firing rate can be modified directly in `HlsMetadataHandler` based on preference

File: docs/android/v2/advanced-features/hls-timed-metadata.mdx [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+});
+
+```
+</Tab>
+
+## Receiving HLS Timed Metadata
+
+To receive metadata on the HLS player side use `HlsMetadataHandler` class from sample app.
+
+```kotlin
```
Comment (saikatmitra91) [inline docs/android/v2/advanced-features/hls-timed-metadata.mdx:66]: Let's document how to sync metadata with player time as well.

**Review by saikatmitra91 — DISMISSED** [review]

File: docs/android/v2/advanced-features/hls-timed-metadata.mdx [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+```java
+
+HMSHLSTimedMetadata metaDataModel = new HMSHLSTimedMetadata(payload,duration,metaData);
+ArrayList<HMSHLSTimedMetadata> metadataModelList = new ArrayList<HMSHLSTimedMetadata>(metaDataModel);
+metadataModelList.add(metaDataModel);
+hmsSDK.setHlsSessionMetadata(metadataModelList, new HMSActionResultListener() {
+    @Override
+    public void onSuccess() {
+        Log.d(TAG, "hls metadata sending failed");
```
Comment (PratimMallick) [inline docs/android/v2/advanced-features/hls-timed-metadata.mdx:50]: @jashan1498 the log says `"hls metadata sending failed");` in `onSuccess`

File: docs/android/v2/advanced-features/hls-timed-metadata.mdx [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    @Override
+    public void onError(@NonNull HMSException e) {
+        Log.d(TAG, "hls metadata sent successfully");
```
Comment (PratimMallick) [inline docs/android/v2/advanced-features/hls-timed-metadata.mdx:55]: Reverse this

File: docs/android/v2/advanced-features/hls-timed-metadata.mdx (hunk 7 above)
Comment (PratimMallick) [reply docs/android/v2/advanced-features/hls-timed-metadata.mdx:66]: @jashan1498 provide a code snippet here

Comment (jashan1498) [reply docs/android/v2/advanced-features/hls-timed-metadata.mdx:69]: resolved

File: docs/android/v2/advanced-features/hls-timed-metadata.mdx (hunk 8 above)
Comment (jashan1498) [reply docs/android/v2/advanced-features/hls-timed-metadata.mdx]: resolved

File: docs/android/v2/advanced-features/hls-timed-metadata.mdx (hunk 9 above)
Comment (jashan1498) [reply docs/android/v2/advanced-features/hls-timed-metadata.mdx]: resolved

**Review by PratimMallick — APPROVED** [review]

**Pull request merged**